enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -0,0 +1,394 @@
+for their Ingress resources so that they can use [Amazon Certificate
+Manager](https://aws.amazon.com/certificate-manager/)(ACM) to manage
+certificates for the Ingress domains and also attach [Web Application
+Firewalls](https://aws.amazon.com/waf/) to the ALB for security and traffic
+shaping. The Openshift [router](github.com/openshift/router) uses an NLB which
+passes through all traffic to the router. TLS termination is done on the router.
+A WAF cannot be attached to the NLB and the router cannot be configured to
+perform TLS termination at the NLB.
+
+## Background
+
+### AWS Load Balancing
+AWS has three kinds of load balancers:
+1. **Classic Load Balancer:** This is a Level 4 load balancer with some Level 7
+    features. It does not have integration with
+    AWS’s newer services.
+2. **Network Load Balancer:** This is also a Level 4 load balancer but has been
+    optimized for performance and cost compared to the Classic Load Balancer.
+3. **Application Load Balancer:** This is a Level 7 load balancer which can
+    route requests based on the HTTP path and host. It also integrates with other
+    AWS services like DDoS protection and WAF.
+
+### Openshift Router
+In OCP/OKD Ingress resources are serviced by the Openshift
+[router](https://github.com/openshift/router). Routers are created by the
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator/)(CIO)
+for every
+_[IngressController](https://docs.openshift.com/container-platform/4.9/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-ingress-controller.html)_
+resource. By default on AWS the CIO creates a Service of type
+[LoadBalancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
+The in-tree Service controller in turn provisions a Network Load
+Balancer with a TCP listener. The CIO then creates a DNS entry for the subdomain
+specified in the IngressController resource with the load balancer created in
+the previous step as the target. 
+
+### aws-load-balancer-controller
+AWS maintains an out-of-tree controller with support for ALBs for Ingress and
+Service type resources. The in-tree Service controller only provisions [classic
+load
+balancers](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/introduction.html)
+and
+[Network](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html)
+Load Balancers. The out-of-tree controller supports integration of Kubernetes
+resources with more of the AWS services like WAFs and the DDoS prevent service
+[Shield](https://docs.aws.amazon.com/waf/latest/developerguide/ddos-overview.html).
+It also enables traffic routing from the load balancer directly to the pods of
+an Ingress or Service if the cluster uses the [AWS VPC Container Networking
+Interface](https://github.com/aws/amazon-vpc-cni-k8s). In addition it also
```

Comment:
Is support for amazon-vpc-cni-k8s in scope or out of scope?

Reply:
Yes, it's out of scope. OCP doesn't offer it as an option so we cannot detect if it is enabled and then enable it in the controller.